android/anc/src/main/java/org/smartregister/fhircore/anc/ui/report/ReportHomeActivity.kt

```diff
@@ -17,29 +17,344 @@
+  private lateinit var fhirEngine: FhirEngine
+  lateinit var parser: IParser
+  lateinit var fhirContext: FhirContext
+  lateinit var libraryEvaluator: LibraryEvaluator
+  lateinit var measureEvaluator: MeasureEvaluator
+  lateinit var fileUtil: FileUtil
+  lateinit var libraryResources: List<IBaseResource>
+  var libraryData: String? = ""
+  var helperData: String? = ""
+  var valueSetData: String? = ""
+  val evaluatorId = "ANCRecommendationA2"
+  val contextCQL = "patient"
+  val contextLabel = "mom-with-anemia"
+  var cqlBaseURL = ""
+  var libraryURL = ""
+  var measureEvaluateLibraryURL = ""
+  var measureTypeURL = ""
+  var cqlMeasureReportURL = ""
+  var cqlMeasureReportStartDate = ""
+  var cqlMeasureReportEndDate = ""
+  var cqlMeasureReportReportType = ""
+  var cqlMeasureReportSubject = ""
+  var cqlMeasureReportLibInitialString = ""
+  var cqlHelperURL = ""
+  var valueSetURL = ""
+  var patientURL = ""
+  var cqlConfigFileName = "configs/cql_configs.properties"
+  lateinit var dir: File
+  lateinit var libraryMeasure: IBaseBundle
+  var measureEvaluateLibraryData: String? = ""
+  lateinit var valueSetBundle: IBaseBundle
+  val dirCQLDirRoot = "cql_libraries"
+  val fileNameMainLibraryCql = "main_library_cql"
+  val fileNameHelperLibraryCql = "helper_library_cql"
+  val fileNameValueSetLibraryCql = "value_set_library_cql"
+  val fileNameMeasureLibraryCql = "measure_library_cql"
+  var patientResourcesIBase = ArrayList<IBaseResource>()
+  lateinit var patientDataIBase: IBaseBundle
+  lateinit var patientDetailsData: String
+  lateinit var patientId: String
+  lateinit var reportViewModel: ReportViewModel
+
+  val executor = Executors.newSingleThreadExecutor()
+  val handler = Handler(Looper.getMainLooper())
+
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
+    libraryEvaluator = LibraryEvaluator()
+    measureEvaluator = MeasureEvaluator()
+    fhirContext = FhirContext.forCached(FhirVersionEnum.R4)
+    parser = fhirContext.newJsonParser()
+    fhirResourceDataSource = FhirResourceDataSource.getInstance(AncApplication.getContext())
+    fhirEngine = AncApplication.getContext().fhirEngine
+
     val patientId =
       intent.extras?.getString(QuestionnaireActivity.QUESTIONNAIRE_ARG_PATIENT_KEY) ?: ""
     val repository = ReportRepository((application as AncApplication).fhirEngine, patientId)
-    val viewModel = ReportViewModel.get(this, application as AncApplication, repository)
-    viewModel.backPress.observe(
+    val dispatcher: DispatcherProvider = DefaultDispatcherProvider
+    reportViewModel = ReportViewModel(repository, dispatcher)
+    reportViewModel.backPress.observe(
       this,
       {
         if (it) {
           finish()
         }
       }
     )
-    setContent { AppTheme { ReportHomeScreen(viewModel) } }
+
+    fileUtil = FileUtil()
+    cqlBaseURL =
+      this.let { fileUtil.getProperty("smart_register_base_url", it, cqlConfigFileName) }!!
```

Comment:
suggestion "smart_register_base_url" and properties below can declare as constants